numba/cpython/setobj.py

```diff
@@ -1089,6 +1094,9 @@ def _copy_payload(self, src_payload):
                     cgutils.raw_memcpy(builder, payload.entries,
                                        src_payload.entries, nentries,
                                        entry_size)
+                    with src_payload._iterate() as loop:
```

Comment:
Shouldn't `meminfo_new_varsize` above be replaced with `meminfo_new_varsize_dtor` similar to the original PR?